src/NHibernate/AdoNet/ConnectionManager.cs

```diff
@@ -259,16 +258,31 @@ private void AggressiveRelease()
 			}
 		}
 
+		[NonSerialized]
+		private int _flushDepth;
+
 		public void FlushBeginning()
 		{
-			log.Debug("registering flush begin");
-			isFlushing = true;
+			if (_flushDepth == 0)
+			{
+				log.Debug("registering flush begin");
```

Comment:
No log on Hibernate side for this. For minimizing logs, I issue it only when the releases flag is toggled. If we would like to use it for tracking begin/end mismatches, it then needs to be moved.